In [2]:
# Import dependencies

import json
import pandas as pd
import numpy as np

In [3]:
file_dir= 'C://Users/lizja/Desktop/Bootcamp/Pre Work/Module 8- ETL/'

f'{file_dir}Movies_ETL'


'C://Users/lizja/Desktop/Bootcamp/Pre Work/Module 8- ETL/Movies_ETL'

In [4]:
with open (f'{file_dir}/wikipedia-movies.json', mode='r') as file:
    wiki_movies_raw=json.load(file)

In [5]:
len(wiki_movies_raw)

7311

In [6]:
# First 5 movies:
wiki_movies_raw[:5]

[{'url': 'https://en.wikipedia.org/wiki/The_Adventures_of_Ford_Fairlane',
  'year': 1990,
  'imdb_link': 'https://www.imdb.com/title/tt0098987/',
  'title': 'The Adventures of Ford Fairlane',
  'Directed by': 'Renny Harlin',
  'Produced by': ['Steve Perry', 'Joel Silver'],
  'Screenplay by': ['David Arnott', 'James Cappe', 'Daniel Waters'],
  'Story by': ['David Arnott', 'James Cappe'],
  'Based on': ['Characters', 'by Rex Weiner'],
  'Starring': ['Andrew Dice Clay',
   'Wayne Newton',
   'Priscilla Presley',
   'Lauren Holly',
   'Morris Day',
   'Robert Englund',
   "Ed O'Neill"],
  'Narrated by': 'Andrew "Dice" Clay',
  'Music by': ['Cliff Eidelman', 'Yello'],
  'Cinematography': 'Oliver Wood',
  'Edited by': 'Michael Tronick',
  'Productioncompany ': 'Silver Pictures',
  'Distributed by': '20th Century Fox',
  'Release date': ['July 11, 1990', '(', '1990-07-11', ')'],
  'Running time': '102 minutes',
  'Country': 'United States',
  'Language': 'English',
  'Budget': '$20 million',


In [7]:
# LastFirst 5 movies:
wiki_movies_raw[-5:]

[{'url': 'https://en.wikipedia.org/wiki/Holmes_%26_Watson',
  'year': 2018,
  'imdb_link': 'https://www.imdb.com/title/tt1255919/',
  'title': 'Holmes & Watson',
  'Directed by': 'Etan Cohen',
  'Produced by': ['Will Ferrell',
   'Adam McKay',
   'Jimmy Miller',
   'Clayton Townsend'],
  'Screenplay by': 'Etan Cohen',
  'Based on': ['Sherlock Holmes',
   'and',
   'Dr. Watson',
   'by',
   'Sir Arthur Conan Doyle'],
  'Starring': ['Will Ferrell',
   'John C. Reilly',
   'Rebecca Hall',
   'Rob Brydon',
   'Steve Coogan',
   'Ralph Fiennes'],
  'Music by': 'Mark Mothersbaugh',
  'Cinematography': 'Oliver Wood',
  'Edited by': 'Dean Zimmerman',
  'Productioncompanies ': ['Columbia Pictures',
   'Gary Sanchez Productions',
   'Mosaic Media Group',
   'Mimran Schur Pictures'],
  'Distributed by': 'Sony Pictures Releasing',
  'Release date': ['December 25, 2018',
   '(',
   '2018-12-25',
   ')',
   '(United States)'],
  'Running time': '90 minutes',
  'Country': 'United States',
  'Language

In [8]:
# Random records in the middle:
wiki_movies_raw[2700:2705]

[{'url': 'https://en.wikipedia.org/wiki/The_Art_of_War_(film)',
  'year': 2000,
  'imdb_link': 'https://www.imdb.com/title/tt0160009/',
  'title': 'The Art of War',
  'Directed by': 'Christian Duguay',
  'Produced by': 'Nicolas Clermont',
  'Screenplay by': ['Wayne Beach', 'Simon Barry'],
  'Story by': 'Wayne Beach',
  'Starring': ['Wesley Snipes',
   'Anne Archer',
   'Maury Chaykin',
   'Marie Matiko',
   'Cary-Hiroyuki Tagawa',
   'Michael Biehn',
   'Donald Sutherland'],
  'Music by': 'Normand Corbeil',
  'Cinematography': 'Pierre Gill',
  'Edited by': 'Michel Arcand',
  'Productioncompany ': ['Morgan Creek',
   'Franchise Pictures',
   'Amen-Ra Films'],
  'Distributed by': 'Warner Bros.',
  'Release date': ['August 25, 2000', '(', '2000-08-25', ')'],
  'Running time': '117 minutes',
  'Country': ['Canada', '[1]', 'United States', '[1]'],
  'Language': 'English',
  'Budget': '$60 million',
  'Box office': '$40.4 million'},
 {'url': 'https://en.wikipedia.org/wiki/Autumn_in_New_York_

In [9]:
kaggle_metadata=pd.read_csv(f'{file_dir}movies_metadata.csv', low_memory=False)
ratings=pd.read_csv(f'{file_dir}ratings.csv')

In [10]:
kaggle_metadata.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

In [11]:
kaggle_metadata.tail()

adult belongs_to_collection budget  \
45461  False                   NaN      0   
45462  False                   NaN      0   
45463  False                   NaN      0   
45464  False                   NaN      0   
45465  False                   NaN      0   

                                                  genres  \
45461  [{'id': 18, 'name': 'Drama'}, {'id': 10751, 'n...   
45462                      [{'id': 18, 'name': 'Drama'}]   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   
45464                                                 []   
45465                                                 []   

                                   homepage      id    imdb_id  \
45461  http://www.imdb.com/title/tt6209470/  439050  tt6209470   
45462                                   NaN  111109  tt2028550   
45463                                   NaN   67758  tt0303758   
45464                                   NaN  227506  tt0008536   
45465                                   NaN  461257  tt6980792   

      original_language       original_title  \
45461                fa              رگ خواب   
45462                tl  Siglo ng Pagluluwal   
45463                en             Betrayal   
45464                en  Satana likuyushchiy   
45465                en             Queerama   

                                                overview  ... release_date  \
45461        Rising and falling between a man and woman.  ...          NaN   
45462  An artist struggles to finish his work while a...  ...   2011-11-17   
45463  When one of her hits goes wrong, a professiona...  ...   2003-08-01   
45464  In a small town live two brothers, one a minis...  ...   1917-10-21   
45465  50 years after decriminalisation of homosexual...  ...   2017-06-09   

      revenue runtime                          spoken_languages    status  \
45461     0.0    90.0    [{'iso_639_1': 'fa', 'name': 'فارسی'}]  Released   
45462     0.0   360.0         [{'iso_639_1': 'tl', 'name': ''}]  Released   
45463     0.0    90.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   
45464     0.0    87.0                                        []  Released   
45465     0.0    75.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                          tagline                title  video  \
45461  Rising and falling between a man and woman               Subdue  False   
45462                                         NaN  Century of Birthing  False   
45463                      A deadly game of wits.             Betrayal  False   
45464                                         NaN     Satan Triumphant  False   
45465                                         NaN             Queerama  False   

      vote_average vote_count  
45461          4.0        1.0  
45462          9.0        3.0  
45463          3.8        6.0  
45464          0.0        0.0  
45465          0.0        0.0  

[5 rows x 24 columns]

In [12]:
kaggle_metadata.sample(5)

adult                              belongs_to_collection    budget  \
5347   False  {'id': 1006, 'name': 'Austin Powers Collection...  63000000   
36539  False                                                NaN   4000000   
30671  False                                                NaN         0   
27571  False                                                NaN         0   
4739   False                                                NaN         0   

                                                  genres  \
5347   [{'id': 35, 'name': 'Comedy'}, {'id': 80, 'nam...   
36539                     [{'id': 27, 'name': 'Horror'}]   
30671                     [{'id': 28, 'name': 'Action'}]   
27571                [{'id': 99, 'name': 'Documentary'}]   
4739                       [{'id': 80, 'name': 'Crime'}]   

                                     homepage      id    imdb_id  \
5347             http://www.austinpowers.com/     818  tt0295178   
36539  http://www.blumhouse.com/tag/the-veil/  257087  tt3533916   
30671                                     NaN   59404  tt0086304   
27571                                     NaN   58587  tt0454121   
4739                                      NaN   27092  tt0227277   

      original_language               original_title  \
5347                 en  Austin Powers in Goldmember   
36539                en                     The Veil   
30671                zh           Shao Lin dou La Ma   
27571                es            Diario de un skin   
4739                 es                Plata quemada   

                                                overview  ... release_date  \
5347   The world's most shagadelic spy continues his ...  ...   2002-07-26   
36539  The story springs from the real-world headline...  ...   2016-04-21   
30671  A kung fu fanatic, Yu Ting (Alexander Lo Rei),...  ...   1983-01-01   
27571                                                NaN  ...   2005-06-13   
4739   Set in Argentina in 1965, the story follows th...  ...   2000-05-11   

           revenue runtime                                   spoken_languages  \
5347   296655431.0    94.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
36539          0.0    93.0           [{'iso_639_1': 'en', 'name': 'English'}]   
30671          0.0    90.0               [{'iso_639_1': 'zh', 'name': '普通话'}]   
27571          0.0    80.0           [{'iso_639_1': 'es', 'name': 'Español'}]   
4739           0.0   125.0           [{'iso_639_1': 'es', 'name': 'Español'}]   

         status                                            tagline  \
5347   Released  He's still evil... He's still deadly... and he...   
36539  Released                                 The dead come back   
30671  Released                                                NaN   
27571  Released                                                NaN   
4739   Released  Based on a true story that for 35 years was ke...   

                             title  video vote_average vote_count  
5347   Austin Powers in Goldmember  False          5.9      997.0  
36539                     The Veil  False          4.4      123.0  
30671             Shaolin vs. Lama  False          7.7        3.0  
27571            Diario de un skin  False          6.7        7.0  
4739                   Burnt Money  False          6.4       21.0  

[5 rows x 24 columns]

In [13]:
ratings.head()

userId  movieId  rating   timestamp
0       1      110     1.0  1425941529
1       1      147     4.5  1425942435
2       1      858     5.0  1425941523
3       1     1221     5.0  1425941546
4       1     1246     5.0  1425941556

In [14]:
ratings.tail()

userId  movieId  rating   timestamp
26024284  270896    58559     5.0  1257031564
26024285  270896    60069     5.0  1257032032
26024286  270896    63082     4.5  1257031764
26024287  270896    64957     4.5  1257033990
26024288  270896    71878     2.0  1257031858

In [15]:
ratings.sample(5)

userId  movieId  rating   timestamp
6803965    70262      364     4.0   836861515
23303956  241863      173     3.0   834428434
16588251  172414     3811     4.0  1000108744
4131194    42950     1103     3.5  1181184251
8259134    85165    44195     3.5  1329954766

In [16]:
wiki_movies_df=pd.DataFrame(wiki_movies_raw)

In [17]:
wiki_movies_df.head()

url    year  \
0  https://en.wikipedia.org/wiki/The_Adventures_o...  1990.0   
1  https://en.wikipedia.org/wiki/After_Dark,_My_S...  1990.0   
2   https://en.wikipedia.org/wiki/Air_America_(film)  1990.0   
3    https://en.wikipedia.org/wiki/Alice_(1990_film)  1990.0   
4      https://en.wikipedia.org/wiki/Almost_an_Angel  1990.0   

                               imdb_link                            title  \
0  https://www.imdb.com/title/tt0098987/  The Adventures of Ford Fairlane   
1  https://www.imdb.com/title/tt0098994/             After Dark, My Sweet   
2  https://www.imdb.com/title/tt0099005/                      Air America   
3  https://www.imdb.com/title/tt0099012/                            Alice   
4  https://www.imdb.com/title/tt0099018/                  Almost an Angel   

          Directed by                  Produced by  \
0        Renny Harlin   [Steve Perry, Joel Silver]   
1         James Foley  [Ric Kidney, Robert Redlin]   
2  Roger Spottiswoode               Daniel Melnick   
3         Woody Allen              Robert Greenhut   
4        John Cornell                 John Cornell   

                                Screenplay by                     Story by  \
0  [David Arnott, James Cappe, Daniel Waters]  [David Arnott, James Cappe]   
1                [James Foley, Robert Redlin]                          NaN   
2                  [John Eskow, Richard Rush]                          NaN   
3                                         NaN                          NaN   
4                                         NaN                          NaN   

                                            Based on  \
0                        [Characters, by Rex Weiner]   
1  [the novel, After Dark, My Sweet, by, Jim Thom...   
2             [Air America, by, Christopher Robbins]   
3                                                NaN   
4                                                NaN   

                                            Starring  ... Predecessor  \
0  [Andrew Dice Clay, Wayne Newton, Priscilla Pre...  ...         NaN   
1  [Jason Patric, Rachel Ward, Bruce Dern, George...  ...         NaN   
2  [Mel Gibson, Robert Downey Jr., Nancy Travis, ...  ...         NaN   
3  [Alec Baldwin, Blythe Danner, Judy Davis, Mia ...  ...         NaN   
4        [Paul Hogan, Elias Koteas, Linda Kozlowski]  ...         NaN   

  Founders Area served Products Services Russian Hebrew Revenue  \
0      NaN         NaN      NaN      NaN     NaN    NaN     NaN   
1      NaN         NaN      NaN      NaN     NaN    NaN     NaN   
2      NaN         NaN      NaN      NaN     NaN    NaN     NaN   
3      NaN         NaN      NaN      NaN     NaN    NaN     NaN   
4      NaN         NaN      NaN      NaN     NaN    NaN     NaN   

  Operating income Polish  
0              NaN    NaN  
1              NaN    NaN  
2              NaN    NaN  
3              NaN    NaN  
4              NaN    NaN  

[5 rows x 193 columns]

In [18]:
wiki_movies_df.tail()

url    year  \
7306    https://en.wikipedia.org/wiki/Holmes_%26_Watson  2018.0   
7307     https://en.wikipedia.org/wiki/Vice_(2018_film)  2018.0   
7308  https://en.wikipedia.org/wiki/On_the_Basis_of_Sex  2018.0   
7309  https://en.wikipedia.org/wiki/Destroyer_(2018_...  2018.0   
7310  https://en.wikipedia.org/wiki/Black_Mirror:_Ba...  2018.0   

                                  imdb_link                title  \
7306  https://www.imdb.com/title/tt1255919/      Holmes & Watson   
7307  https://www.imdb.com/title/tt6266538/                 Vice   
7308  https://www.imdb.com/title/tt4669788/  On the Basis of Sex   
7309  https://www.imdb.com/title/tt7137380/            Destroyer   
7310  https://www.imdb.com/title/tt9495224/         Bandersnatch   

       Directed by                                        Produced by  \
7306    Etan Cohen  [Will Ferrell, Adam McKay, Jimmy Miller, Clayt...   
7307    Adam McKay  [Brad Pitt, Dede Gardner, Jeremy Kleiner, Kevi...   
7308    Mimi Leder                                     Robert W. Cort   
7309  Karyn Kusama             [Fred Berger, Phil Hay, Matt Manfredi]   
7310   David Slade                                     Russell McLean   

     Screenplay by Story by  \
7306    Etan Cohen      NaN   
7307           NaN      NaN   
7308           NaN      NaN   
7309           NaN      NaN   
7310           NaN      NaN   

                                               Based on  \
7306  [Sherlock Holmes, and, Dr. Watson, by, Sir Art...   
7307                                                NaN   
7308                                                NaN   
7309                                                NaN   
7310                                                NaN   

                                               Starring  ... Predecessor  \
7306  [Will Ferrell, John C. Reilly, Rebecca Hall, R...  ...         NaN   
7307  [Christian Bale, Amy Adams, Steve Carell, Sam ...  ...         NaN   
7308  [Felicity Jones, Armie Hammer, Justin Theroux,...  ...         NaN   
7309  [Nicole Kidman, Sebastian Stan, Toby Kebbell, ...  ...         NaN   
7310  [Fionn Whitehead, Will Poulter, Craig Parkinso...  ...         NaN   

     Founders Area served Products Services Russian Hebrew Revenue  \
7306      NaN         NaN      NaN      NaN     NaN    NaN     NaN   
7307      NaN         NaN      NaN      NaN     NaN    NaN     NaN   
7308      NaN         NaN      NaN      NaN     NaN    NaN     NaN   
7309      NaN         NaN      NaN      NaN     NaN    NaN     NaN   
7310      NaN         NaN      NaN      NaN     NaN    NaN     NaN   

     Operating income Polish  
7306              NaN    NaN  
7307              NaN    NaN  
7308              NaN    NaN  
7309              NaN    NaN  
7310              NaN    NaN  

[5 rows x 193 columns]

In [19]:
wiki_movies_df.sample (10)

url    year  \
2166             https://en.wikipedia.org/wiki/Star_Kid  1997.0   
5566     https://en.wikipedia.org/wiki/Pearl_Jam_Twenty  2011.0   
808                                                 NaN     NaN   
4437  https://en.wikipedia.org/wiki/The_Texas_Chains...  2006.0   
4282  https://en.wikipedia.org/wiki/Kill_Your_Darlin...  2006.0   
1540          https://en.wikipedia.org/wiki/Brain_Candy  1996.0   
479        https://en.wikipedia.org/wiki/Bad_Lieutenant  1992.0   
1016  https://en.wikipedia.org/wiki/How_the_West_Was...  1994.0   
2952      https://en.wikipedia.org/wiki/Delivering_Milo  2001.0   
3855       https://en.wikipedia.org/wiki/Cinderella_Man  2005.0   

                                  imdb_link  \
2166  https://www.imdb.com/title/tt0120478/   
5566  https://www.imdb.com/title/tt1417592/   
808                                     NaN   
4437  https://www.imdb.com/title/tt0420294/   
4282  https://www.imdb.com/title/tt0472118/   
1540  https://www.imdb.com/title/tt0116768/   
479   https://www.imdb.com/title/tt0103759/   
1016  https://www.imdb.com/title/tt0110068/   
2952  https://www.imdb.com/title/tt0212910/   
3855  https://www.imdb.com/title/tt0352248/   

                                            title         Directed by  \
2166                                     Star Kid          Manny Coto   
5566                             Pearl Jam Twenty       Cameron Crowe   
808                                           NaN                 NaN   
4437  The Texas Chainsaw Massacre:  The Beginning  Jonathan Liebesman   
4282                           Kill Your Darlings       Björne Larson   
1540                Kids in the Hall: Brain Candy         Kelly Makin   
479                                Bad Lieutenant        Abel Ferrara   
1016                         How the West Was Fun     Stuart Margolin   
2952                              Delivering Milo         Nick Castle   
3855                               Cinderella Man          Ron Howard   

                                            Produced by  \
2166                                  Jennie Lew Tugend   
5566   [Michele Anthony, Cameron Crowe, Morgan Neville]   
808                                                 NaN   
4437  [Michael Bay, Mike Fleiss, Tobe Hooper, Kim He...   
4282                                                NaN   
1540                                     Lorne Michaels   
479                                  Edward R. Pressman   
1016                                                NaN   
2952  [Heidi Levitt, Deepak Nayar, Heriberto Schoeffer]   
3855         [Ron Howard, Penny Marshall, Brian Grazer]   

                              Screenplay by                          Story by  \
2166                                    NaN                               NaN   
5566                                    NaN                               NaN   
808                                     NaN                               NaN   
4437                         Sheldon Turner  [Sheldon Turner, David J. Schow]   
4282                                    NaN                               NaN   
1540                                    NaN                               NaN   
479                                     NaN                               NaN   
1016                                    NaN                               NaN   
2952                                    NaN                               NaN   
3855  [Cliff Hollingsworth, Akiva Goldsman]               Cliff Hollingsworth   

                                               Based on  \
2166                                                NaN   
5566                                                NaN   
808                                                 NaN   
4437  [The Texas Chainsaw Massacre, by Tobe Hooper, ...   
4282                                                NaN   
1540                                                NaN   
479                                  

### Since there are 193 columns, let's turn check out the columns :
Notice not all the columns print out down below:



In [21]:
wiki_movies_df.columns

Index(['url', 'year', 'imdb_link', 'title', 'Directed by', 'Produced by',
       'Screenplay by', 'Story by', 'Based on', 'Starring',
       ...
       'Predecessor', 'Founders', 'Area served', 'Products', 'Services',
       'Russian', 'Hebrew', 'Revenue', 'Operating income', 'Polish'],
      dtype='object', length=193)

### Let us view it as a list 

In [22]:
wiki_movies_df.columns.tolist()

['url',
 'year',
 'imdb_link',
 'title',
 'Directed by',
 'Produced by',
 'Screenplay by',
 'Story by',
 'Based on',
 'Starring',
 'Narrated by',
 'Music by',
 'Cinematography',
 'Edited by',
 'Productioncompany ',
 'Distributed by',
 'Release date',
 'Running time',
 'Country',
 'Language',
 'Budget',
 'Box office',
 'Written by',
 'Genre',
 'Theme music composer',
 'Country of origin',
 'Original language(s)',
 'Producer(s)',
 'Editor(s)',
 'Production company(s)',
 'Original network',
 'Original release',
 'Productioncompanies ',
 'Executive producer(s)',
 'Production location(s)',
 'Distributor',
 'Picture format',
 'Audio format',
 'Voices of',
 'Followed by',
 'Composer(s)',
 'Created by',
 'Also known as',
 'Opening theme',
 'No. of episodes',
 'Preceded by',
 'Author',
 'Publisher',
 'Publication date',
 'Media type',
 'Pages',
 'ISBN',
 'OCLC',
 'LC Class',
 'Cover artist',
 'Series',
 'Set in',
 'Adaptation by',
 'Suggested by',
 'Biographical data',
 'Born',
 'Died',
 'Resti

In [24]:
wiki_movies= [movie for movie in wiki_movies_raw
              if ('Director' in movie or 'Directed by' in movie) 
              and 'imdb_link' in movie
             and 'No. of episodes' not in movie]

len(wiki_movies)

7076

In [25]:
sorted(wiki_movies_df.columns.tolist())

['Actor control',
 'Adaptation by',
 'Alias',
 'Alma mater',
 'Also known as',
 'Animation by',
 'Arabic',
 'Area',
 'Area served',
 'Artist(s)',
 'Attraction type',
 'Audio format',
 'Author',
 'Based on',
 'Biographical data',
 'Bopomofo',
 'Born',
 'Box office',
 'Budget',
 'Camera setup',
 'Cantonese',
 'Characters',
 'Children',
 'Chinese',
 'Cinematography',
 'Closing date',
 'Color process',
 'Comics',
 'Composer(s)',
 'Coordinates',
 'Country',
 'Country of origin',
 'Cover artist',
 'Created by',
 'Date premiered',
 'Designer(s)',
 'Developed by',
 'Developer(s)',
 'Dewey Decimal',
 'Died',
 'Directed by',
 'Director',
 'Distributed by',
 'Distributor',
 'Divisions',
 'Duration',
 'Edited by',
 'Editor(s)',
 'Ending theme',
 'Engine',
 'Engine(s)',
 'Executive producer(s)',
 'Family',
 'Fate',
 'Film(s)',
 'Followed by',
 'Format(s)',
 'Formerly',
 'Founded',
 'Founder',
 'Founders',
 'French',
 'Full name',
 'Gender',
 'Genre',
 'Genre(s)',
 'Genres',
 'Gwoyeu Romatzyh',
 'Ha

In [31]:
def clean_movie(movie):
    movie=dict(movie) # Create a non-destructive copy
    alt_titles = {}
    for key in ['Also known as','Arabic','Cantonese','Chinese','French',
                'Hangul','Hebrew','Hepburn','Japanese','Literally',
                'Mandarin','McCune–Reischauer','Original title','Polish',
                'Revised Romanization','Romanized','Russian',
                'Simplified','Traditional','Yiddish']:
        if key in movie:
            alt_titles[key] = movie[key]
            movie.pop(key)
            
    if len(alt_titles)>0:
        movie['alt_titles']=alt_titles
    return movie
    

In [33]:
clean_movies = [clean_movie(movie) for movie in wiki_movies]
wiki_movies_df = pd.DataFrame(clean_movies)
sorted(wiki_movies_df.columns.tolist())

['Adaptation by',
 'Animation by',
 'Audio format',
 'Based on',
 'Box office',
 'Budget',
 'Cinematography',
 'Color process',
 'Composer(s)',
 'Country',
 'Country of origin',
 'Created by',
 'Directed by',
 'Director',
 'Distributed by',
 'Distributor',
 'Edited by',
 'Editor(s)',
 'Executive producer(s)',
 'Followed by',
 'Genre',
 'Label',
 'Language',
 'Length',
 'Music by',
 'Narrated by',
 'Original language(s)',
 'Original network',
 'Original release',
 'Picture format',
 'Preceded by',
 'Produced by',
 'Producer',
 'Producer(s)',
 'Production company(s)',
 'Production location(s)',
 'Productioncompanies ',
 'Productioncompany ',
 'Recorded',
 'Release date',
 'Released',
 'Running time',
 'Screen story by',
 'Screenplay by',
 'Starring',
 'Story by',
 'Suggested by',
 'Theme music composer',
 'Venue',
 'Voices of',
 'Written by',
 'alt_titles',
 'imdb_link',
 'title',
 'url',
 'year']